### ISSUE TITLE
Problems with foreign characters in urls

### ISSUE THREAD

**Issue by mskogly** [stevenvachon/broken-link-checker#53]
Installed and ran blc on http://poetikon.no. Getting a HTTP_404  on a link containing a Norwegian special character "ø". http://poetikon.no/category/gj%C3%B8r-det-selv/   (http://bit.ly/2eRQv52 if the urlencoded url fails).

This url works when visiting in a browser.

**Comment by stevenvachon** [comment]
Try using the v0.8.0 branch

**Comment by agross** [comment]
Unfortunately this branch fails for me:

```sh
$ node node_modules/broken-link-checker/bin/blc -fvr http://beta.grossweber.com/test.html
Getting links from: http://beta.grossweber.com/test.html
(node:11168) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'length' of undefined
(node:11168) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by matkoniecz** [comment]
For me it fails on "ą" and gives false 400 error code - other link checkers are not complaining about a working link.

I scanned readme and I failed to see method to check just a single link (to futher debug whatever my guess that I triggered this error is correct).

For reference, full link is https://matkoniecz.github.io/bicycle_map_of_Krakow/Przegląd%20tras%20rowerowych.html and at this time it is working (note this site is experimental and url may disappear soon).

```
wget https://matkoniecz.github.io/bicycle_map_of_Krakow/Przegląd%20tras%20rowerowych.html > test.html
--2017-01-31 19:47:13--  https://matkoniecz.github.io/bicycle_map_of_Krakow/Przegl%C4%85d%20tras%20rowerowych.html
Resolving matkoniecz.github.io (matkoniecz.github.io)... 151.101.112.133
Connecting to matkoniecz.github.io (matkoniecz.github.io)|151.101.112.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1648 (1,6K) [text/html]
Saving to: ‘Przegl\304%85d tras rowerowych.html’

100%[===================================================================================================================================================================================================>] 1.648       --.-K/s   in 0s      

2017-01-31 19:47:14 (212 MB/s) - ‘Przegl\304%85d tras rowerowych.html’ saved [1648/1648]
```

**Comment by stevenvachon** [comment]
A tool that only checks a single link is pretty useless. But if you must, you can use `UrlChecker.prototype.enqueue`. `v0.7.x` of this library uses Node's `url.parse()`, which does not follow the new URL specification. `v0.8.0` does, but is not yet released.

**Comment by matkoniecz** [comment]
> A tool that only checks a single link is pretty useless.

Well, for debygging what is causing false results it would be quite useful.

**Comment by stevenvachon** [comment]
For which there is probably more than one, hence a tool for multiple. Single URLs are still possible with that API.

**Comment by stevenvachon** [comment]
I think that this is fixed (on master branch).
